kimu/core/intersection_tool_line_circle.py

```diff
@@ -0,0 +1,272 @@
+        # https://www.cuemath.com/geometry/equation-of-circle/
+        # for more information.
+        # 3. Search for intersection point of these two functions by
+        # analytically modifying the resulting equation so that it is
+        # possible to solve x (and then y). Note that we end up with
+        # quadratic equation ->> we will end up with two possible
+        # solutions. The only exceptions are that the selected line
+        # does not intersect with the circle at all or that the line
+        # acts as a tangent for the circle.
+        a = (
+            (line_coords[3]) ** 2.0
+            - 2.0 * line_coords[1] * line_coords[3]
+            + (line_coords[1]) ** 2.0
+            + (line_coords[2]) ** 2.0
+            - 2.0 * line_coords[0] * line_coords[2]
+            + (line_coords[0]) ** 2.0
+        )
+        # print(f"Value of a is {a}")
+        b = (
+            -2.0 * (line_coords[3]) ** 2.0 * line_coords[0]
+            + 2.0 * line_coords[1] * line_coords[3] * line_coords[2]
+            + 2.0 * line_coords[1] * line_coords[3] * line_coords[0]
+            - 2.0 * (line_coords[1]) ** 2.0 * line_coords[2]
+            - 2.0 * centroid[0] * (line_coords[2]) ** 2.0
+            - 2.0 * centroid[0] * (line_coords[0]) ** 2.0
+            + 4.0 * centroid[0] * line_coords[0] * line_coords[2]
+            - 2.0 * line_coords[2] * centroid[1] * line_coords[3]
+            + 2.0 * centroid[1] * line_coords[1] * line_coords[2]
+            + 2.0 * centroid[1] * line_coords[3] * line_coords[0]
+            - 2.0 * centroid[1] * line_coords[1] * line_coords[0]
+        )
+        # print(f"Value of b is {b}")
+        c = (
+            (line_coords[3]) ** 2.0 * (line_coords[0]) ** 2.0
+            - 2.0 * line_coords[0] * line_coords[1] * line_coords[2] * line_coords[3]
+            + (line_coords[1]) ** 2.0 * (line_coords[2]) ** 2.0
+            + (centroid[0]) ** 2.0 * (line_coords[2]) ** 2.0
+            - 2.0 * (centroid[0]) ** 2.0 * line_coords[0] * line_coords[2]
+            + (centroid[0]) ** 2.0 * (line_coords[0]) ** 2.0
+            + 2.0 * line_coords[2] * centroid[1] * line_coords[3] * line_coords[0]
+            - 2.0 * (line_coords[2]) ** 2.0 * centroid[1] * line_coords[1]
+            - 2.0 * (line_coords[0]) ** 2.0 * centroid[1] * line_coords[3]
+            + 2.0 * centroid[1] * line_coords[1] * line_coords[2] * line_coords[0]
+            + (line_coords[2]) ** 2.0 * (centroid[1]) ** 2.0
+            - (line_coords[2]) ** 2.0 * r ** 2.0
+            - 2.0 * line_coords[0] * line_coords[2] * (centroid[1]) ** 2.0
+            + 2.0 * line_coords[0] * line_coords[2] * r ** 2.0
+            + (line_coords[0]) ** 2.0 * (centroid[1]) ** 2.0
+            - (line_coords[0]) ** 2.0 * r ** 2.0
+        )
+        # print(f"Value of c is {c}")
+
+        # Check that the selected line feature and indirectly
+        # defined circle intersect.
+        # To do: Complains that there is no intersection point
```

Comment:
```suggestion
        # TODO: fix intersection tool for small radius values
        # Complains that there is no intersection point
```

Reply:
I do not why but my code-style check failed if I used TODO and that is why I didn't use it..

Reply:
Nevertheless, this problem gets resolved in the following commits to this branch